Zend/zend_gc.c

```diff
@@ -1555,6 +1555,8 @@ ZEND_API int zend_gc_collect_cycles(void)
 						GC_ADD_FLAGS(obj, IS_OBJ_FREE_CALLED);
 						GC_ADDREF(obj);
 						obj->handlers->free_obj(obj);
+						current = GC_IDX2PTR(idx);  // bug #78811: free_obj() can cause the root buffer to be reallocated.
+						obj = (zend_object*)GC_GET_PTR(current->ref);
```

Comment:
On second thought, reassigning `obj` might be the unnecessary part.

`					current->ref = GC_MAKE_GARBAGE(((char*)obj) - obj->handlers->offset);` - The left hand assignment was to the pointer `current`, which is the expression that should be recomputed.